src/library/components/ProductCard.tsx

```diff
@@ -0,0 +1,163 @@
+  const { tag, texts, headingLevel, linkTo, variant, image, gradientBrand, dropShadow} = props;
+
+  const [clonedImage, setClonedImage] = useState<ReactElement>();
+  const [clonedTag, setClonedTag] = useState<ReactElement>();
+
+  const productNameId = useId();
+
+  let cardStyles = {
+    classes: "",
+    titleLinkClasses: "",
+    linkClasses: ""
+  }
+
+  const defaultStyle = {
+    classes: "bsds-card",
+    titleLinkClasses: "bsds-link-subtle",
+    linkClasses: ""
+  }
+
+  const ghostStyle = {
+    classes: "bsds-card-ghost",
+    titleLinkClasses: "bsds-link-ghost",
+    linkClasses: "bsds-link-ghost"
+  }
+
+  const borderLightStyle = {
+    classes: "bsds-card-border-light",
+    titleLinkClasses: "bsds-link-subtle",
+    linkClasses: ""
+  }
+
+  const borderGhostStyle = {
+    classes: "bsds-card-border-ghost",
+    titleLinkClasses: "bsds-link-ghost",
+    linkClasses: "bsds-link-ghost"
+  }
+
+  switch(variant) {
+    case "ghost":
+      cardStyles = {...ghostStyle};
+      break;
+    case "border-light":
+      cardStyles = {...borderLightStyle};
+      break;
+    case "border-ghost":
+      cardStyles = {...borderGhostStyle};
+      break;
+
+    default:
+      cardStyles = {...defaultStyle};
+      break;
+  }
+
+  let decorativeState = "";
+  if (gradientBrand && variant?.includes("ghost")) {
+    decorativeState = "bsds-card-gradient";
+    cardStyles = {...borderGhostStyle};
+  } else if (gradientBrand) {
+    decorativeState = "bsds-card-gradient";
+    cardStyles = {...borderLightStyle};
+  } else if (dropShadow && variant?.includes("ghost")) {
+    decorativeState = "bsds-card-shadow";
+    cardStyles = {...borderGhostStyle};
+  } else if (dropShadow) {
+    decorativeState = "bsds-card-shadow";
+    cardStyles = {...borderLightStyle};
+  }
+
+  useEffect(() => {
+    if(tag) {
+      setClonedTag(cloneElement(tag as ReactElement, {
+        isGhost: (variant === "ghost" || variant === "border-ghost"),
+     }));
+    }
+  },[tag, variant])
+
+  useEffect(() => {
+    if(image) {
+      setClonedImage(cloneElement(image as ReactElement, {
+        ratio: (image.props.ratio ? image.props.ratio : "1:1"),
+        isDecorative: false,
+        isGhost: (variant === "ghost" || variant === "border-ghost"),
+     }));
+    }
+  },[image, linkTo, variant])
+
+  useEffect(() => {
+    //This is to expand the link hitbox for 50% split image
+    if(image?.props.ratio === "50:50") {
+      const productCardLinks = document.querySelectorAll(".bsds-product-card-with-image");
+      Array.prototype.forEach.call(productCardLinks, productCardLink => {
+        let keyDown:number, keyUp:number, link = productCardLink.querySelector("a");
+        productCardLink.onmousedown = () => keyDown = +new Date();
+        productCardLink.onmouseup = () => {
+          keyUp = +new Date();
+          if((keyUp - keyDown) < 200) {
+            link.click();
+          }
+        }
+      });
+    }
+  }, [image?.props.ratio])
```

Comment:
Is the goal to make the title text and image clickable and not the description? If so, can we put the onmouseup event listener on just the img element
`img = productCardLink.querySelector('img')`
and remove the timing constraint?

Would it be possible to wrap the image in the <a>? Or does this method allow us to reuse more styles?

Reply:
This has been updated to add the onmouseup event listener onto image element only.